src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs

```diff
@@ -208,8 +210,28 @@ private static void VerifyKey_RSA(RSA rsa)
 
             byte[] expectedExponent = new byte[] { 0x01, 0x00, 0x01 };
 
-            Assert.Equal(expectedModulus, rsaParameters.Modulus);
-            Assert.Equal(expectedExponent, rsaParameters.Exponent);
+            if (!expectedModulus.SequenceEqual(rsaParameters.Modulus) ||
+                !expectedExponent.SequenceEqual(rsaParameters.Exponent))
+            {
+                rsaParameters = rsa.ExportParameters(false);
+
+                if (!expectedModulus.SequenceEqual(rsaParameters.Modulus) ||
+                    !expectedExponent.SequenceEqual(rsaParameters.Exponent))
+                {
+                    Assert.False(true, "Second call to ExportParameters produced valid data, but the first did not");
+                }
+
+                rsa = cert.GetRSAPublicKey();
+                rsaParameters = rsa.ExportParameters(false);
+
+                if (!expectedModulus.SequenceEqual(rsaParameters.Modulus) ||
+                    !expectedExponent.SequenceEqual(rsaParameters.Exponent))
+                {
+                    Assert.False(true, "New key handle ExportParameters was successful, original key handle was not");
+                }    
+
+                Assert.False(true, "Key's busted");
```

Comment:
Might need a better message here :).  Maybe even the original Assert.Equal (though maybe not, so we can tell we have the new logging?)